jsonnet/kube-prometheus/platforms/kops.libsonnet

```diff
@@ -18,22 +18,22 @@ local service(name, namespace, labels, selector, ports) = {
     kubeControllerManagerPrometheusDiscoveryService: service(
       'kube-controller-manager-prometheus-discovery',
       'kube-system',
-      { 'app.kubernetes.io/name': 'kube-controller-manager' },
-      { 'app.kubernetes.io/name': 'kube-controller-manager' },
+      { 'k8s-app': 'kube-controller-manager' },
```

Comment:
To allow smooth operations, this line should contain both labels, `app.kubernetes.io/name` and `k8s-app` as it is passed to metadata and not to the selector.

Reply:
Here we are passing parameters to a `service()` function. This function accepts 5 parameters and cannot be extended by appending new ones. This should be written as (I added comments to explain what is each parameter doing):
```jsonnet
kubeControllerManagerPrometheusDiscoveryService: service(
  'kube-controller-manager-prometheus-discovery',  # name of the Service object
  'kube-system',  # namespace in which Service object is placed
  { 'k8s-app': 'kube-controller-manager', 'app.kubernetes.io/name': 'kube-controller-manager' }  # labels added as metadata to the Service object
  { 'k8s-app': 'kube-controller-manager' },  # labels used in Service selector
  [{ name: 'https-metrics', port: 10257, targetPort: 10257 }]  # list of ports for the Service
),
```

The above example will create a Service with 2 labels that will match Pods with `k8s-app: kube-controller-manager` label set.

Pattern like this should be extended to the other Services from this addon.

Reply:
Oh this was stupid mistake, fixed it.